crates/runtime/src/datafusion.rs

```diff
@@ -122,6 +123,7 @@ impl DataFusion {
     pub async fn new_accelerated_backend(
         &self,
         dataset: impl Borrow<Dataset>,
+        secrets_provider: Arc<RwLock<secrets::SecretsProvider>>,
```

Comment:
nit: I think we can read backend secret before calling `new_accelerated_backend`, and pass it down instead of whole secrets provider

Reply:
Updated, thanks!